cfn-resources/util/util.go

```diff
@@ -50,6 +57,38 @@ func CreateMongoDBClient(publicKey, privateKey string) (*mongodbatlas.Client, er
 	return atlas, nil
 }
 
+func NewMongoDBClient(req handler.Request) (*mongodbatlas.Client, *handler.ProgressEvent) {
+	keys, handlerError := getApiKeys(req)
+	if handlerError != nil {
+		return nil, handlerError
+	}
+
+	// Create atlas client
+	client, err := CreateMongoDBClient(keys.PublicKey, keys.PrivateKey)
+	if err != nil {
+		_, _ = cfnlog.Warnf("Create - error: %+v", err)
+		peErr := progress_events.GetFailedEventByCode(fmt.Sprintf("Error creating mongoDB client : %s", err.Error()),
+			cloudformation.HandlerErrorCodeInvalidRequest)
+		return nil, &peErr
+	}
+
+	return client, nil
+}
+
+func getApiKeys(req handler.Request) (*DeploymentSecret, *handler.ProgressEvent) {
+	key, err := GetAPIKeyFromDeploymentSecret(&req, "Atlas-Cloud-Formation-ApiKey-Secret")
```

Comment:
[q] Should "Atlas-Cloud-Formation-ApiKey-Secret" be user configured? How do we expose this information to users so they know which secret to configure?